models/error.go

```diff
@@ -1925,6 +1925,21 @@ func (err ErrReviewNotExist) Error() string {
 	return fmt.Sprintf("review does not exist [id: %d]", err.ID)
 }
 
+// ErrIllLegalReviewRequest The review request doesn't follow permission rules.
+type ErrIllLegalReviewRequest struct {
+	Msg string
+}
+
+// IsErrIllLegalReviewRequest checks if an error is a ErrIllLegalReviewRequest.
+func IsErrIllLegalReviewRequest(err error) bool {
+	_, ok := err.(ErrIllLegalReviewRequest)
+	return ok
+}
+
+func (err ErrIllLegalReviewRequest) Error() string {
+	return err.Msg
+}
+
```

Comment:
Illegal not IllLegal

```suggestion
// ErrIllegalReviewRequest The review request doesn't follow permission rules.
type ErrIllegalReviewRequest struct {
	Msg string
}

// IsErrIllegalReviewRequest checks if an error is a ErrIllegalReviewRequest.
func IsErrIllegalReviewRequest(err error) bool {
	_, ok := err.(ErrIllegalReviewRequest)
	return ok
}

func (err ErrIllegalReviewRequest) Error() string {
	return err.Msg
}

```

Reply:
haha this could be written by me too :D